python_modules/libraries/dagster-looker/dagster_looker/asset_decorator.py

```diff
@@ -39,72 +40,73 @@ def build_looker_explore_specs(project_dir: Path) -> Sequence[AssetSpec]:
     return looker_explore_specs
 
 
-def build_looker_view_specs(project_dir: Path) -> Sequence[AssetSpec]:
-    looker_view_specs = []
+def build_asset_key_from_sqlglot_table(table: exp.Table) -> AssetKey:
+    return AssetKey([part.name.replace("*", "_star") for part in table.parts])
+
+
+def parse_upstream_asset_keys_from_looker_view(
+    looker_view: Mapping[str, Any], looker_view_path: Path
+) -> AbstractSet[AssetKey]:
     sql_dialect = "bigquery"
 
-    # https://cloud.google.com/looker/docs/reference/param-view
-    for view_path in project_dir.rglob("*.view.lkml"):
-        for view in lkml.load(view_path.read_text()).get("views", []):
-            upstream_tables: Sequence[exp.Table] = [to_table(view["name"], dialect=sql_dialect)]
-
-            # https://cloud.google.com/looker/docs/derived-tables
-            derived_table_sql = view.get("derived_table", {}).get("sql")
-
-            if derived_table_sql and "$" in derived_table_sql:
-                logger.warn(
-                    f"Failed to parse the derived table SQL for view `{view['name']}`"
-                    f" in file {view_path.name}, because the SQL in this view contains the"
-                    " LookML substitution operator, `$`."
-                    " The upstream dependencies for the view will be omitted."
-                )
+    # We need to handle the Looker substitution operator ($) properly since the lkml
+    # compatible SQL may not be parsable yet by sqlglot.
+    #
+    # https://cloud.google.com/looker/docs/sql-and-referring-to-lookml#substitution_operator_
+    upstream_view_pattern = re.compile(r"\${(.*?)\.SQL_TABLE_NAME\}")
```

Comment:
fussy nit but could move this down to be directly adjacent to the first code that uses it on line 68